libr/include/passmanager.h

```diff
@@ -0,0 +1,73 @@
+/* radare2 - LGPL - Copyright 2019 - oddcoder*/
+
+#ifndef R2_PASSMANAGER_H
+#define R2_PASSMANAGER_H
+
+#include <r_util.h>
+#include <r_anal.h>
+
+typedef enum {
+        ModulePass, //output of rbin
+        FunctionPass, // function 
+        BBPass, //basic block
+	//TODO path for path analysis
+} PassType;
+
+typedef struct pm PassManager;
+typedef struct pass Pass;
+/*
+ * Always add elements to this structure at the bottom not at the begining,
+ * so it is always backward compatiable
+ */
+
+typedef struct pass{
+        char *name; //must be unique for every pass;
+        PassType t;
+        void (*registerDependencies)(PassManager *);
+	void *(*run)(PassManager *, Pass *p, void *object);
+	void (*invalidate)(PassManager *, Pass *p, void *object);
+	void *customDataStructure;
+	//TODO  passes can come with their own set of commands
+} Pass;
+
+#ifdef PASSMANAGER_IMPLEMENTATION_FOR_INTERNAL_USE_ONLY
+
+
+typedef struct pm{
+	bool log;
+	PassType t;
+	Sdb *passes; //(k,v) = (Pass->name, PassRunner) 
```

Comment:
Just use `HtPP` please.

Reply:
HtPP is for pointer pointer hashtable but the key here is string, Am I missing something?

Reply:
And what is a string if not a pointer to an array of chars? :D

See this test https://github.com/radare/sdb/blob/master/test/unit/test_hash.c#L251 to understand how to use it.